TradeItIosTicketSDK2/TradeItLinkedBrokerManager.swift

```diff
@@ -268,20 +268,34 @@ import PromiseKit
         return self.linkedBrokers.filter { $0.error == nil }
     }
 
-    public func unlinkBroker(_ linkedBroker: TradeItLinkedBroker) {
-        self.connector.unlinkLogin(linkedBroker.linkedLogin, localOnly: false)
-
-        if let index = self.linkedBrokers.index(of: linkedBroker), let userId = linkedBroker.linkedLogin.userId {
-            TradeItSDK.linkedBrokerCache.remove(linkedBroker: linkedBroker)
-            self.linkedBrokers.remove(at: index)
-            self.oAuthDelegate?.didUnlink?(userId: userId)
-            NotificationCenter.default.post(
-                name: TradeItSDK.didUnlinkNotificationName,
-                object: nil,
-                userInfo: [
-                    "linkedBroker": linkedBroker
-                ]
-            )
+    public func unlinkBroker(_ linkedBroker: TradeItLinkedBroker,
+                             onSuccess: @escaping () -> Void,
+                             onFailure: @escaping (TradeItErrorResult) -> Void) {
+        self.connector.unlinkLogin(linkedBroker.linkedLogin, localOnly: false) { result in
+            switch result {
+            case _ as TradeItUnlinkLoginResult:
+                if let index = self.linkedBrokers.index(of: linkedBroker), let userId = linkedBroker.linkedLogin.userId {
+                    TradeItSDK.linkedBrokerCache.remove(linkedBroker: linkedBroker)
+                    self.linkedBrokers.remove(at: index)
+                    self.oAuthDelegate?.didUnlink?(userId: userId)
+                    NotificationCenter.default.post(
+                        name: TradeItSDK.didUnlinkNotificationName,
+                        object: nil,
+                        userInfo: [
+                            "linkedBroker": linkedBroker
+                        ]
+                    )
+                    onSuccess()
+                } else {
+                    let error = TradeItErrorResult(title: "UnlinkBroker error")
```

Comment:
I think the `title` is shown to the user so we should give it a nicer title. There might also be `TradeItErrorResult.systemError` helper.